core/src/main/java/org/infinispan/commands/tx/PrepareCommand.java

```diff
@@ -68,21 +69,14 @@
 
    protected WriteCommand[] modifications;
    protected boolean onePhaseCommit;
-   protected CacheNotifier notifier;
-   protected RecoveryManager recoveryManager;
    private transient boolean replayEntryWrapping  = false;
    protected boolean retriedCommand;
 
    private static final WriteCommand[] EMPTY_WRITE_COMMAND_ARRAY = new WriteCommand[0];
 
    @Override
-   public void init(ComponentRegistry componentRegistry, boolean isRemote) {
-      super.init(componentRegistry, isRemote);
-      this.notifier = componentRegistry.getCacheNotifier().running();
-      this.recoveryManager = componentRegistry.getRecoveryManager().running();
-
-      for (ReplicableCommand nested : getModifications())
-         componentRegistry.getCommandsFactory().initializeReplicableCommand(nested, false);
+   public void init(ComponentRegistry componentRegistry) {
+      markTransactionAsRemote(true);
```

Comment:
`init()` is not going to be called any more, unless you call it from `invokeAsync`.

TBH I don't think we really need ':local' and ':remote' in `GlobalTransaction.toString()`, but that's for another PR.